platform/migration/platform-migratable/src/main/java/org/codice/ddf/migration/impl/PlatformMigratable.java

```diff
@@ -0,0 +1,216 @@
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.migration.impl;
+
+import static org.apache.commons.lang.Validate.notNull;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Collection;
+
+import javax.validation.constraints.NotNull;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.lang.StringUtils;
+import org.codice.ddf.migration.Migratable;
+import org.codice.ddf.migration.MigrationException;
+import org.codice.ddf.migration.MigrationMetadata;
+import org.codice.ddf.migration.MigrationWarning;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class handles the export process for all Platform configurations and system files.
+ *
+ */
+public class PlatformMigratable implements Migratable {
+
+    private static final Logger LOGGER = LoggerFactory
+            .getLogger(PlatformMigratable.class);
+
+    private static final String KEYSTORE_SYSTEM_PROP = "javax.net.ssl.keyStore";
+
+    private static final String TRUSTSTORE_SYSTEM_PROP = "javax.net.ssl.trustStore";
+
+    private static final String WS_SECURITY_DIR = "etc/ws-security";
+
+    private static final String SYSTEM_PROPERTIES = "etc/system.properties";
+
+    private static final String USERS_PROPERTIES = "etc/users.properties";
+    
+    private static final String ABSOLUTE_PATH_WARNING =
+            "The value for property [%s] is set to a path [%s] that is absolute; "
+                    + "therefore, the file will not be included in the export.  "
+                    + "Check that the file exists on the system you're migrating to "
+                    + "or update the property value and export again.";
+
+    private static final String OUTSIDE_PATH_WARNING =
+            "The value for property [%s] is set to a path [%s] that is outside [%s]; "
+                    + "therefore, the file will not be included in the export.  "
+                    + "Check that the file exists on the system you're migrating to "
+                    + "or update the property value and export again.";
+
+    private static final String UNREAL_PATH_WARNING =
+            "The value for property [%s] is set to a path [%s] that could not coerced into a real path; "
+                    + "therefore, the file will not be included in the export.  "
+                    + "Check that the file exists on the system you're migrating to "
+                    + "or update the property value and export again.";
+
+    private final Path ddfHome;
+    
+    public PlatformMigratable(@NotNull Path ddfHome) {
+        notNull(ddfHome, "ddfHome cannot be null");
+        this.ddfHome = ddfHome;
+    }
+    
+    @Override
```

Comment:
Is this an implemented method of an interface? There is no `@override` annotation on this interface (nor any documentation). If the `@NotNull` annotation will only live on the interfaces, it would sure be nice to have a javadoc link to that method so it would be easily discernable that this methods parameters have a specific contract.. maybe i'm strange but it feels like its a lost piece of information if its not included in anyway on the implementation.